openedx/core/djangoapps/credentials/tasks/v1/tasks.py

```diff
@@ -54,3 +62,178 @@ def send_grade_to_credentials(self, username, course_run_key, verified, letter_g
             f"Failed to send grade to credentials. Reason: {error_msg}"
         )
         raise self.retry(exc=exception, countdown=countdown, max_retries=MAX_RETRIES)
+
+
+@shared_task(bind=True, ignore_result=True)
+@set_code_owner_attribute
+def handle_notify_credentials(self, options, course_keys):
+    """
+    Celery task to handle the notify_credentials management command. Finds the
+    relevant cert and grade records, then starts other celery tasks to send the
+    data.
+    """
+
+    try:
+        site_config = SiteConfiguration.objects.get(site__domain=options['site']) if options['site'] else None
+    except SiteConfiguration.DoesNotExist:
+        logger.error('No site configuration found for site %s', options['site'])
+
+    certs = get_recently_modified_certificates(
+        course_keys, options['start_date'], options['end_date'], options['user_ids']
+    )
+
+    users = None
+    if options['user_ids']:
+        users = User.objects.filter(id__in=options['user_ids'])
+
+    grades = get_recently_modified_grades(
+        course_keys, options['start_date'], options['end_date'], users
+    )
+
+    logger.info('notify_credentials Sending notifications for {certs} certificates and {grades} grades'.format(
+        certs=certs.count(),
+        grades=grades.count()
+    ))
+
+    if options['dry_run']:
+        print_dry_run(certs, grades)
+    else:
+        send_notifications(
+            certs,
+            grades,
+            site_config=site_config,
+            delay=options['delay'],
+            page_size=options['page_size'],
+            verbose=options['verbose'],
+            notify_programs=options['notify_programs']
+        )
+
+    logger.info('notify_credentials finished')
+
+
+def send_notifications(
+    certs, grades, site_config=None, delay=0, page_size=100, verbose=False, notify_programs=False
+):
+    """ Run actual handler commands for the provided certs and grades. """
+
+    # Imported here instead of above to avoid circular import
+    from openedx.core.djangoapps.credentials.signals import send_grade_if_interesting
```

Comment:
This is new. That `signals` file imports the `send_grade_to_credentials` task above, so if this is up at the top, it causes problems because a partially imported file is trying to import another partially imported file. [This page ](https://stackabuse.com/python-circular-imports/)led me to believe that this solution is okay (as well as other places around the codebase where I’ve seen importing deferred like this).